core/src/main/java/org/bitcoinj/core/VersionMessage.java

```diff
@@ -98,7 +98,7 @@ public VersionMessage(NetworkParameters params, byte[] payload) throws ProtocolE
     public VersionMessage(NetworkParameters params, int newBestHeight) {
         super(params);
         clientVersion = params.getProtocolVersionNum(NetworkParameters.ProtocolVersion.CURRENT);
-        localServices = 0;
+        localServices = NODE_WITNESS;
```

Comment:
I think this is wrong, we're not a full node in the sense that we can answer queries for blocks and transactions.